spec/classes/augeasproviders_instances_spec.rb

```diff
@@ -1,30 +1,33 @@
+# frozen_string_literal: true
+
 require 'spec_helper'
 
 describe 'augeasproviders::instances' do
-  it { should create_class('augeasproviders::instances') }
-  it { should contain_class('augeasproviders::params') }
+  it { is_expected.to create_class('augeasproviders::instances') }
+  it { is_expected.to contain_class('augeasproviders::params') }
 
   context "when resource_defaults => 'foo'" do
-    let(:params) {{ :resource_defaults => 'foo' }}
-    it { expect { should create_class('augeasproviders::instances') }.to raise_error(Puppet::Error, /is not a Hash/) }
+    let(:params) { { resource_defaults: 'foo' } }
+
+    it { expect { is_expected.to create_class('augeasproviders::instances') }.to raise_error(Puppet::Error, %r{expects a Hash value, got String}) }
```

Comment:
I know it was a problem before, but the native way is to do this:
```suggestion
    it { is_expected.to compile.and_raise_error(%r{expects a Hash value, got String}) }
```